README.md

```diff
@@ -162,23 +159,132 @@ searchBoxPlaceHolder: "Search",
 
 ### Algorithm parameters
 
-The standard website pipeline defines several algorithm parameters. For example, `resultsPerPage`.
+The standard website pipeline defines several algorithm parameters. For example, `q` or `resultsPerPage`.
 
 ```javascript
 values: {
-   resultsPerPage: "10", // Show 10 results per page.
+  q: getUrlParam("q") // The initial search query will be the value of the query param "q".
+  resultsPerPage: "10", // Show 10 results per page.
 },
 ```
 
-### Initial Values
+### Events
 
-These values described in this section are only added to the value map when the app starts.
-If the overlay is closed or the query is cleared, these values will be removed.
+You can subscribe to events by calling your interface with the `"sub"` value followed by the event name and then a callback.
 
 ```javascript
-initialValues: {
-  q: getUrlParam("q") // The initial search query will be the value of the query param "q".
+myUI("sub", "<event>", function() {});
+```
+
+| Event | Data | Description |
+| :-- | :-: | :-- |
+| `"search-sent"` | value dictionary | Search request has been sent |
+| `"values-changed"` | value dictionary | Value map has changed |
+| `"response-updated"` | response object | Response has updated |
+| `"page-close"` | query string | Page is about to be closed |
+| `"query-reset"` | query string | Body has changed enough to be considered a new query |
+| `"result-clicked"` | query string | Result has been clicked |
+| `"search-event"` | query string | Search event |
+| `"overlay-show"` | none | Overlay is shown |
+| `"overlay-hide"` | none | Overlay is hidden |
+
+You can also publish events which the search interface will pick up.
+
+| Event | Data | Description |
+| :-- | :-: | :-- |
+| `"values-set"` | value dictionary | Values to set |
+| `"search-send"` | none | Perform a search |
+| `"overlay-show"` | none | Show the overlay |
+| `"overlay-hide"` | none | Hide the overlay |
+
+#### Search Sent
+
+```javascript
+myUI("sub", "search-sent", function(eventName, values) {
+  console.log("Search sent with ", values);
+});
+```
+
+#### Values Changed
+
+```javascript
+myUI("sub", "values-changed", function(eventName, values) {
+  console.log("New values are", values);
+});
+```
+
+#### Response Updated
+
+```javascript
+myUI("sub", "response-updated", function(eventName, response) {
+  if (response.isEmpty()) {
+    return;
+  }
+  if (response.isError()) {
+    console.log("Got error", response.getError());
+  } else {
+    console.log("Got results", response.getResults());
+  }
+});
+```
+
+#### Search Event
```

Comment:
This needs work.  I would start the other way around, with `analytics-search-event` and the say that it can be triggered by 3 intermediary events, which can also be observed by attaching to their names etc...